.github/workflows/ci.yaml

```diff
@@ -0,0 +1,70 @@
+on: [push, pull_request]
+
+name: squakey
+
+jobs:
```

Comment:
I don't configure GitHub Actions that often, but have I understand correctly that this config runs `cargo check`, `cargo test` and `rustfmt` &  `clippy` as three separate concurrent jobs? I find it a bit questionable as there's quite a lot of overlap: running tests requires compiling the code which accomplishes the same as `cargo check`, and IIRC `clippy` reports `cargo check` warnings alongside its own lints.

Personally I would drop the `check` job because I think it's redundant, but I'm open to good arguments.

Reply:
Cool, I'll squash them into the same. Just picked a template to have something.